mbedtls/src/ecp/mod.rs

```diff
@@ -373,9 +369,14 @@ Please use `mul_with_rng` instead."
     ///
     /// This function will return an error if:
     ///
-    /// * `k` is not a valid private key, or `self` is not a valid public key.
+    /// * `k` is not a valid private key, determined by mbedtls function [`mbedtls_ecp_check_privkey`]
```

Comment:
I again think it does not make sense to speak about private/public keys here. It's just scalars and EC points.

But, OK, these functions is also in upstream under `ecp.c`.

For the nitpicking's sake, I suggest
```
The scalar `k` is not valid as a private key
```
and 
```
The point is not valid as a public key
```
maybe more clear.

Reply:
Updated!